react/features/toolbox/components/web/OverflowMenuButton.tsx

```diff
@@ -29,46 +34,83 @@ interface IProps {
     /**
      * A child React Element to display within {@code Popover}.
      */
-    children: ReactNode;
+    children: Object[];
```

Comment:
Why do you need to change this type?

Reply:
Because now we are not constructing the JSX outside of the OverflowMenuButton but inside it.

Reply:
Not sure I understand. What type are the children now? I haven't seen that change.

Reply:
They are array with objects. Those are the grouped objects which are representing the buttons from where we transform them in JSX. Check Toolbox.tsx. Before we were building the whole JSX for the content of the overflow menu. I've moved the JSX building here.

Reply:
In Toolbox, the OverflowMenuButton currently only has one child: ContextMenu. And it's definitely JSX, not an Object

Reply:
@robertpin See this change. https://github.com/jitsi/jitsi-meet/pull/13211/files#diff-3e1fb4316fab0334bc37d37a0f046777e7ce3fd8c8e4321cc8951edbffe2b972L1432 . IMHO this is definitely an array of Objects.

Am I missing something?

Reply:
Oh, I didn't see you changed it. Hmm, seems really weird to pass some Objects as children. Why do we do this?

Reply:
I don't see why it is weird to pass Objects as children. Could you please explain?

Correct me if this is wrong but what I've been told (or read) the `children` prop is like any other prop, isn't it? I've seen it widely used for all kind of types (in our project and in another projects as well).

The reason for the change is that I need to put the ReactionsMenu as a child of the ContextMenu that we are building on one side and on the other to not over-complicate the Toolbox component which have been already too big and complicated. Also IMHO I don't think that we need to keep the OverflowMenuButton so generic. Now we pass to it what needs to be rendered and it takes care how to render it.

Reply:
Now I'm passing the object array with the information about the buttons into normal prop (not children).